crates/node/src/network_server/grpc_svc_handler.rs

```diff
@@ -279,4 +284,98 @@ impl NodeCtlSvc for NodeCtlSvcHandler {
 
         Ok(Response::new(cluster_state_response))
     }
+
+    /// Get a versioned kv-pair
+    async fn metadata_get(
+        &self,
+        request: tonic::Request<MetadataGetRequest>,
+    ) -> Result<Response<MetadataGetResponse>, Status> {
+        let request = request.into_inner();
+        let value = self
+            .metadata_store_client
+            .inner()
+            .get(request.key.into())
+            .await
+            .map_err(|err| Status::internal(err.to_string()))?;
```

Comment:
If we wanted to maintain information on whether the error is retryable or not, then we probably need to encode it differently. Should probably depend on whether the user of this API `restatectl` would retry or not. Probably we don't do this right now. So then this should be ok.

Reply:
right now `restatectl` will automatically try a different node until it succeeds or exhaust the list of available nodes. But I get why it is useful to include the "retryablitiy" of the error in the response.